contentcuration/contentcuration/frontend/shared/views/MarkdownEditor/plugins/registerCustomMarkdownField.js

```diff
@@ -0,0 +1,33 @@
+import Vue from 'vue';
+import vueCustomElement from 'vue-custom-element';
+
+export default VueComponent => {
+  const dashed = camel => camel.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase();
+  const name = dashed(VueComponent.name);
+  Vue.use(vueCustomElement);
+  Vue.customElement(name, VueComponent, {
+    // element is removed from document
+    disconnectedCallback() {
+      this.observer.disconnect();
+    },
+    // Vue instance is created
+    vueInstanceCreatedCallback() {
+      // by default, `contenteditable` will be false
+      this.setAttribute('contenteditable', !!VueComponent.contentEditable);
+
+      // pass innerHTML of host element as the `markdown` property
```

Comment:
Use a `MutationObserver` to watch the contents of the host `span` element.  When the contents change, pass them to the underlying Vue component as a property called `markdown`.